cg/meta/workflow/mutant/quality_controller/report_generator_utils.py

```diff
@@ -0,0 +1,47 @@
+from pathlib import Path
+
+from cg.io.json import write_json
+from cg.meta.workflow.mutant.constants import QUALITY_REPORT_FILE_NAME
+from cg.meta.workflow.mutant.quality_controller.models import (
+    CaseQualityResult,
+    SamplesQualityResults,
+)
+from cg.meta.workflow.mutant.quality_controller.result_logger_utils import samples_results_message
+
+
+def get_report_path(case_path: Path) -> Path:
+    return case_path.joinpath(QUALITY_REPORT_FILE_NAME)
+
+
+def write_report(
+    case_path: Path,
+    case_quality_result: CaseQualityResult,
+    samples_quality_results: SamplesQualityResults,
+) -> None:
+    report_file_path: Path = get_report_path(case_path=case_path)
+
+    summary: str = get_summary(
+        case_quality_result=case_quality_result,
+        samples_quality_results=samples_quality_results,
+        report_file_path=report_file_path,
+    )
+    report_content = {
+        "summary": summary,
+        "case": case_quality_result.model_dump(),
+        "samples": samples_quality_results.model_dump(),
+    }
```

Comment:
What would be the benefit of making this a model? 🤔

Reply:
It will improve the structure, maintainability, and allow validation of this report content as a whole, which is what you want in the end. Also, Pydantic models are self-documenting, so they will help others understand the structure of the report, as you will use type annotations for the case and sample quality results. And we will end up deserialising once, rather than individually for the case and sample results

Reply:
OK, thank you for the clarification! I have added a model now!